CloneTiving/CloneTiving/Views/Main/MainView.swift

```diff
@@ -0,0 +1,200 @@
+        let group = NSCollectionLayoutGroup.horizontal(layoutSize: groupSize, repeatingSubitem: item, count: 4)
+        // section 설정
+        let section = NSCollectionLayoutSection(group: group)
+        
+        //헤더뷰 레이아웃
+        section.boundarySupplementaryItems = [.init(layoutSize: .init(widthDimension: .fractionalWidth(1), heightDimension: .absolute(50)), elementKind: UICollectionView.elementKindSectionHeader, alignment: .topLeading)]
+        //스크롤 설정
+        section.orthogonalScrollingBehavior = .continuous
+        
+        return section
+    }
+    
+    func createBasicLayout() -> NSCollectionLayoutSection {
+        let itemSize = NSCollectionLayoutSize(widthDimension: .absolute(160), heightDimension: .absolute(80))
+        let item = NSCollectionLayoutItem(layoutSize: itemSize)
+        item.contentInsets = NSDirectionalEdgeInsets(top: 4, leading: 4, bottom: 4, trailing: 4)
+        
+        let groupSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .absolute(160.0))
+        let group = NSCollectionLayoutGroup.horizontal(layoutSize: groupSize, repeatingSubitem: item, count: 3)
+        
+        let section = NSCollectionLayoutSection(group: group)
+        
+        section.orthogonalScrollingBehavior = .continuous
+        
+        section.boundarySupplementaryItems = [.init(layoutSize: .init(widthDimension: .fractionalWidth(1), heightDimension: .absolute(50)), elementKind: UICollectionView.elementKindSectionHeader, alignment: .topLeading)]
+        
+        return section
+    }
+    func createLayout() -> UICollectionViewCompositionalLayout {
+        return UICollectionViewCompositionalLayout {[weak self] sectionNumber, env -> NSCollectionLayoutSection? in
+            
+            switch sectionNumber {
+            case 0:
+                return self?.createMainLayout()
+            case 2:
+                return self?.createBasicLayout()
+            default:
+                return self?.createPosterLayout()
+            }
+        }
+    }
+    
+}
+
+extension MainView: UICollectionViewDataSource {
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        5
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        if section == 0 {
+            return largePosterDummyData.count
+        } else if section == 2{
+            return contentDummyData.count
+        } else {
+            return posterDummyData.count
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if indexPath.section == 0 {
+            guard let cell = mainCollectionView.dequeueReusableCell(withReuseIdentifier: LargePosterCollectionViewCell.identifier, for: indexPath) as? LargePosterCollectionViewCell else {
+                return UICollectionViewCell()
```

Comment:
이부분의 코드는 프로토콜을 채택하면
```suggestion
let cell = LargePosterCollectionViewCell.dequeueReusableCell(collectionView: collectionView)
```
로 간단해집니다 그리고 이쪽 코드에서 지희님이 `mainCollectionView`을 그대로 쓰셨는데 함수의 인자로 받아지는거라 `collectionView`라고 쓰셔도됩니다

Reply:
와,, 항상 같은 처리하는게 조금 불편했는데 프로토콜을 이렇게 쓸 수도 있군요.. 정성스러운 코드 리뷰 너무 감사합니다..